indico/util/user.py

```diff
@@ -129,6 +129,23 @@ def principal_from_identifier(identifier, allow_groups=False):
         if user is None:
             raise ValueError('Invalid user: {}'.format(user_id))
         return user
+    elif type_ == 'ExternalUser':
+        if not allow_external_users:
+            raise ValueError('External users are not allowed')
+        cache = GenericCache('external-user')
+        external_user_data = cache.get(data)
+        if not external_user_data:
+            raise ValueError('Invalid data')
+        user = User.query.filter(User.all_emails == external_user_data['email'], ~User.is_deleted).first()
+        if user:
+            return user
+        # create a pending user. this user isn't sent to the DB unless it gets added
+        # to the sqlalchemy session somehow (e.g. by using it to an ACL).
```

Comment:
Did you mean "by adding it to an ACL"?

Reply:
yep, "using it in an ACL" / "adding it to an ACL"